server/core/app.ts

```diff
@@ -33,78 +34,50 @@ export const returnExpress: (
       >,
       { moduleName, addModule }: UIServerModule
     ) => {
-      log(
-        serverName,
-        'startup',
-        'core',
-        'moduleMount',
-        `Mounting module '${moduleName}'`
-      );
+      logger.info(`Mounting module '${moduleName}'`);
       const config = getConfig();
       const { mountPoint, routerForModule } = addModule(
-        (moduleName) => generateLoggers(serverName, moduleName, 'startup'),
+        (moduleName) => generateLogger(moduleName),
         authFunction(config.authentication),
         config
       );
-      log(
-        serverName,
-        'startup',
-        'core',
-        'moduleMount',
-        `Mounted module '${moduleName}' on '${mountPoint}'`
-      );
+      logger.info(`Mounted module '${moduleName}' on '${mountPoint}'`);
       return { ...acc, [moduleName]: { mountPoint, routerForModule } };
     },
     {}
   );
 
   // before all handlers, add context and request start/end handlers
   app.all('*', (req, res, next) => {
-    // add id to request for use downstream
-    if (!req.headers[STRIMZI_UI_REQUEST_ID_HEADER]) {
-      req.headers[STRIMZI_UI_REQUEST_ID_HEADER] = randomBytes(8).toString(
-        'hex'
-      );
-    }
-    const requestID = req.headers[STRIMZI_UI_REQUEST_ID_HEADER] as string;
-    // and make sure the response has the header
-    res.setHeader(STRIMZI_UI_REQUEST_ID_HEADER, requestID);
-    // create a 'context' for this request, containing config, a request ID, and loggers. Available to handlers via `res.locals.strimziuicontext`
+    // set up logger for start/end of request
+    httpLogger(req, res);
```

Comment:
does the httpLogger also handle the end of the request?

Reply:
To check: is this middleware? If so, should it be inserted as a part of the app.all?

Reply:
To check: req.id contains the id, I assume generated in httplogger - should it not be in res.locals? (https://expressjs.com/en/api.html#res.local)

Reply:
Yep - the httpLogger adds the end request logging by default. 
Yes pino-http can be treated as middleware - I will update to use it like that. 
And yep, req.id is added to the request in the pino-http code via the genReqID callback (https://github.com/ajborley/strimzi-ui/commit/52e636b23e35078c1d8c8bbed677d8e143501cac#diff-072fff4cca78a5c9ceb244882beffb3d8970dff40d5134e7c85c9a14b4e1548bR95) - not sure why it isn't added to res.locals - is that an express-only thing? pino-http can be applied to other web frameworks including express.

Reply:
Sorry - forgot to come back and update these having read through everything!

I read from various sources that `res.locals` is the place to put per request state/data. I think they were of the opinion the request should be treated as immutable, but the response was ours to manipulate as we saw fit. Must admit I like that model, hence what I had previously. Thus my preference would be for it to be in res.locals, but happy to see what the consesnus is. All I would say is if we move it/add it, can we update the typings as needed: https://github.com/strimzi/strimzi-ui/blob/ae571ee62875127d08716a42beb1ba9f58f79651/server/types.ts#L143

Reply:
The request ID is still being added to `res.locals.strimziuicontext.requestID` here: https://github.com/ajborley/strimzi-ui/commit/52e636b23e35078c1d8c8bbed677d8e143501cac#diff-7fde445974fe7c9eddd663ca12666ad12830c2800b3e1d8b3d731e8849870ceeR61
It's just generated via pino-http now and added (by pino-http) to the `req` object (which makes sense to me as the request ID should also be treated as immutable, alongside the rest of the request).